libs/community/tests/integration_tests/vectorstores/test_opensearch.py

```diff
@@ -66,6 +66,112 @@ def test_opensearch_with_custom_field_name() -> None:
     assert output == [Document(page_content="foo")]
 
 
+def test_configure_search_pipeline() -> None:
+    """Test configure search pipeline functionality."""
+    test_search_pipeline_name = "test_search_pipeline"
+    keyword_weight = 0.7
+    vector_weight = 0.3
+
+    docsearch = OpenSearchVectorSearch.from_texts(
+        texts, FakeEmbeddings(), opensearch_url=DEFAULT_OPENSEARCH_URL
+    )
+    docsearch.configure_search_pipelines(
+        pipeline_name=test_search_pipeline_name,
+        keyword_weight=keyword_weight,
+        vector_weight=vector_weight,
+    )
+    assert docsearch.search_pipeline_exists(test_search_pipeline_name)
+
+
+def test_get_search_pipeline_info() -> None:
+    """Test get search pipeline info functionality."""
+    test_search_pipeline_name = "test_search_pipeline"
+
+    docsearch = OpenSearchVectorSearch.from_texts(
+        texts, FakeEmbeddings(), opensearch_url=DEFAULT_OPENSEARCH_URL
+    )
+    test_pipeline_info = docsearch.get_search_pipeline_info(test_search_pipeline_name)
+    assert test_pipeline_info == {
+        "test_search_pipeline": {
+            "description": "Post processor for hybrid search",
+            "phase_results_processors": [
+                {
+                    "normalization-processor": {
+                        "normalization": {"technique": "min_max"},
+                        "combination": {
+                            "technique": "arithmetic_mean",
+                            "parameters": {"weights": [0.7, 0.3]},
+                        },
+                    }
+                }
+            ],
+        }
+    }
+
+
+def test_hybrid_search() -> None:
+    """Test hybrid search functionality."""
+    metadatas = [{"page": i} for i in range(len(texts))]
+    docsearch = OpenSearchVectorSearch.from_texts(
+        texts,
+        ConsistentFakeEmbeddings(),
+        metadatas=metadatas,
+        opensearch_url=DEFAULT_OPENSEARCH_URL,
+    )
+    output = docsearch.similarity_search(
+        query="foo",
+        k=2,
+        search_type=HYBRID_SEARCH,
+        search_pipeline="test_search_pipeline",
```

Comment:
this test should be independent of other tests. It shouldn't assume that a test search pipeline exists. Could you implement a clean up step in the tests above, so they don't live a configured pipeline?

Reply:
But to perform the hybrid search. it needs a search_pipeline. if we don't pass the search_pipeline name it will raise an error.

Reply:
I also have another idea to merge both configure_search_pipeline  & hybrid_Search into a single functionality 

where the user directly passes search_type = hybrid_search and weightage for (keyword_search & vector_search) in tuple (0.7, 0.3) -> internally we can check if the search_pipeline already exists. if it exists, it will use it or else it will configure a new search pipeline.